src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -196,32 +196,73 @@ public void GetFullPath_InvalidArgs()
+        public static TheoryData<string, string> GetFullPath_BasicExpansions
         {
-            string curDir = Directory.GetCurrentDirectory();
-            yield return new object[] { curDir, curDir }; // Current directory => current directory
-            yield return new object[] { ".", curDir }; // "." => current directory
-            yield return new object[] { "..", Path.GetDirectoryName(curDir) }; // "." => up a directory
-            yield return new object[] { Path.Combine(curDir, ".", ".", ".", ".", "."), curDir }; // "dir/./././." => "dir"
-            yield return new object[] { curDir + new string(Path.DirectorySeparatorChar, 3) + ".", curDir }; // "dir///." => "dir"
-            yield return new object[] { Path.Combine(curDir, "..", Path.GetFileName(curDir), ".", "..", Path.GetFileName(curDir)), curDir }; // "dir/../dir/./../dir" => "dir"
-            yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "somedir", ".."), Path.GetPathRoot(curDir) }; // "C:\somedir\.." => "C:\"
-            yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "."), Path.GetPathRoot(curDir) }; // "C:\." => "C:\"
-            yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "..", "..", "..", ".."), Path.GetPathRoot(curDir) }; // "C:\..\..\..\.." => "C:\"
-            yield return new object[] { Path.GetPathRoot(curDir) + new string(Path.DirectorySeparatorChar, 3), Path.GetPathRoot(curDir) }; // "C:\\\" => "C:\"
-
-            // Path longer than MaxPath that normalizes down to less than MaxPath
-            const int Iters = 10000;
-            var longPath = new StringBuilder(curDir, curDir.Length + (Iters * 2));
-            for (int i = 0; i < 10000; i++)
+            get
+            {
+                string currentDirectory = Directory.GetCurrentDirectory();
+                string root = Path.GetPathRoot(currentDirectory);
+                string fileName = Path.GetFileName(currentDirectory);
+
+                TheoryData<string, string> data = new TheoryData<string, string>
+                {
+                    // Current directory => current directory
+                    { currentDirectory, currentDirectory },
+                    // "." => current directory
+                    { ".", currentDirectory },
+                    // ".." => up a directory
+                    { "..", Path.GetDirectoryName(currentDirectory) },
+                    // "dir/./././." => "dir"
+                    { Path.Combine(currentDirectory, ".", ".", ".", ".", "."), currentDirectory },
+                    // "dir///." => "dir"
+                    { currentDirectory + new string(Path.DirectorySeparatorChar, 3) + ".", currentDirectory },
+                    // "dir/../dir/./../dir" => "dir"
+                    { Path.Combine(currentDirectory, "..", fileName, ".", "..", fileName), currentDirectory },
+                    // "C:\somedir\.." => "C:\"
```

Comment:
Paths like `C: ...` are treated the same on Linux? I guess it's just string manipulation without regard to the OS?

Reply:
They aren't, it just is using the Windows example. This code is getting the root from the CWD, would be "/" on Unix.

Reply:
Ah yes.